ext/storage_sources/azure_store/azure_connection.cpp

```diff
@@ -95,3 +102,48 @@ azure_connection::get_object(const std::string &path) const
 {
     return 0;
 }
+
+int
+azure_connection::object_exists(const std::string &object_key, bool &exists) const
+{
+    exists = false;
+    std::string obj = _object_prefix + object_key;
+
+    auto list_blob_response = _azure_client.ListBlobs();
+
+    for (const auto blob_item : list_blob_response.Blobs) {
+        // Check if object exists.
+        if (blob_item.Name.compare(obj) == 0) {
+            // Check if object is deleted and has not been cleared by garbage collection.
+            if (blob_item.IsDeleted) {
+                return -1;
+            }
+            exists = true;
```

Comment:
I think there's another logic bug here and below: won't we keep looping even after we find a matching item? And then if there's a second blob where the name matches but it's deleted, we'll return -1. I think we should `break` after the `exists = true`.

Reply:
ty for the suggestion :) implemented

Reply:
Maybe I'm looking at the commit history incorrectly, but - I think you only added it for `bucket_exists` and not `blob_exists`?

Reply:
mb missed `object_exists`, added now